src/D2L.CodeStyle.Analyzers/Diagnostics.cs

```diff
@@ -543,5 +543,15 @@ public static class Diagnostics {
 			defaultSeverity: DiagnosticSeverity.Error,
 			isEnabledByDefault: true
 		);
+
+		public static readonly DiagnosticDescriptor ImplicitUnnamedArgs = new DiagnosticDescriptor(

```

Comment:
Let's consolidate on `NamedArgumentsRequired`:

* as a user I don't really care why they are required, just that they are (i.e. the fix is the same in any case)
* the rules are very complex, actually: e.g. we would never require naming for an argument like `m_foo.Bar` being passed to a parameter named `bar`... we don't want people to have to internalize all these rules because they are uninteresting, so we don't need to explain it in the diagnostics
* the conventions at D2L are very strong and intuitive (and this analyzer does a good job of matching the intuition) so its unlikely anyone would ever be surprised or interested in exactly why an argument was required to be named

That diagnostic has a good title: "Named arguments are required for readability.". The description is a little bit off if we consolidate on it, but it's still pretty applicable.